java/arcs/core/storage/StorageProxy.kt

```diff
@@ -400,7 +401,7 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
 
         when (priorState) {
             ProxyState.AWAITING_SYNC -> notifyReady()
-            ProxyState.SYNC -> notifyUpdate(value)
+            ProxyState.SYNC -> if (valueBefore != value) notifyUpdate(value)
```

Comment:
Does this always work? What if `consumerView` returns an object which is constructed per call?

Reply:
as long as that object is either a data class, or implements `equals` correctly, it should be okay..  I think that holds true for our various Crdt types, as well as RawEntity. I will make some tests to enforce it.  Alternatively, we could deliver both the old and the new `consumerData` objects.